### PR TITLE
Replace logger.Logger with slog

### PR DESCRIPTION
**Pull request by rgrandl**

### CODE REVIEW

File: runtime/logging/logger.go [hunk 1]
```diff
@@ -38,100 +38,98 @@ type Options struct {
...
 	entry := protos.LogEntry{
-		App:        opts.App,
-		Version:    opts.Deployment,
-		Component:  opts.Component,
-		Node:       opts.Weavelet,
+		App:        h.Opts.App,
+		Version:    h.Opts.Deployment,
+		Component:  h.Opts.Component,
+		Node:       h.Opts.Weavelet,
 		TimeMicros: time.Now().UnixMicro(),
```
Comment (mwhittaker) [inline runtime/logging/logger.go:96]: Should we use `rec.Time`?

File: runtime/logging/logger.go [hunk 2]
```diff
@@ -38,100 +38,98 @@ type Options struct {
...
+}
+
+// Enabled implements the slog.Handler interface.
+func (h *LogHandler) Enabled(context.Context, slog.Level) bool {
+	// Support all logging levels.
+	return true
+}
+
+// WithAttrs implements the slog.Handler interface.
+func (h *LogHandler) WithAttrs(attrs []slog.Attr) slog.Handler {
```
Comment (mwhittaker) [inline runtime/logging/logger.go:65]: Is `WithAttrs` supposed to return a copy of the handler and leave the original handler unmodified? For example, consider the following code: 

```go
logger := ...
logger.WithAttrs([]slog.Attry{{"foo", slog.StringValue("bar")}})
logger.Debug("hi")
```

With this code, it looks like the `Debug` log would have the `"foo"="bar"` attribute, but I would expect it not to? I could be misunderstanding the `Handler` interface though.

File: runtime/logging/logger.go [hunk 3]
```diff
@@ -38,100 +38,98 @@ type Options struct {
...
+	for _, attr := range attrs {
+		h.Opts.Attrs = append(h.Opts.Attrs, attr.Key)
+		h.Opts.Attrs = append(h.Opts.Attrs, attr.Value.String())
+	}
+	return h
+}
+
+// WithGroup implements the slog.Handler interface.
+func (h *LogHandler) WithGroup(string) slog.Handler {
+	return h
```
Comment (mwhittaker) [inline runtime/logging/logger.go:77]: Add a TODO to implement this?

**Review by mwhittaker — CHANGES_REQUESTED** [review]
Awesome! I feel like this change simplifies things a lot. :wood::wood::wood:

File: runtime/logging/logger.go [hunk 4]
```diff
@@ -38,100 +38,105 @@ type Options struct {
...
+		Level:      rec.Level.String(),
+		File:       "",
+		Line:       -1,
+		Msg:        rec.Message,
+		Attrs:      appendAttrs(h.Opts.Attrs, attrs),
 	}
-	l.Write(e)
+
+	// Get the file and line information.
+	details := runtime.FuncForPC(rec.PC)
```
Comment (dnephin) [inline runtime/logging/logger.go:112]: I was just reading over the docs for `slog`, and I noticed this about `rec.PC` (https://pkg.go.dev/golang.org/x/exp/slog#Record):

	// The only valid use for this value is as an argument to
	// [runtime.CallersFrames]. In particular, it must not be passed to
	// [runtime.FuncForPC].
	PC [uintptr](https://pkg.go.dev/builtin#uintptr)
	
It doesn't say why, but do you think this call is safe?

In `slog`, the `rec.PC` is used with `runtime.CallersFrames`, which I think can get you the same info:

https://github.com/golang/exp/blob/10a507213a2916eb81243cb95bcd7f5e70a771e1/slog/handler.go#L288

https://github.com/golang/exp/blob/10a507213a2916eb81243cb95bcd7f5e70a771e1/slog/record.go#L70-L74

**Review by dnephin — COMMENTED** [review]
Nice change! I noticed one thing, sorry I was too slow to leave the comment before merge!

File: runtime/logging/logger.go (hunk 4 above)
Comment (rgrandl) [reply runtime/logging/logger.go:112]: Thanks Daniel. Great point.

**Pull request merged**